internal/provider/resource_dashboard_variable.go

```diff
@@ -0,0 +1,267 @@
+	CollectionID                 types.String `tfsdk:"collection_id"`
+	DefaultObjectSelection       types.String `tfsdk:"default_object_selection"`
+	AllowMultipleObjectSelection types.Bool   `tfsdk:"allow_multiple_object_selection"`
+	LastUpdated                  types.String `tfsdk:"last_updated"`
+}
+
+func (r *DashboardVariableResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_dashboard_variable"
+}
+
+func (r *DashboardVariableResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Dashboard variables enable flexible and reusable dashboards by allowing viewers to switch between objects dynamically.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				MarkdownDescription: "The unique identifier of the dashboard variable.",
+				Computed:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"workspace_id": schema.StringAttribute{
+				MarkdownDescription: "The unique identifier of the workspace.",
+				Required:            true,
+			},
+			"collection_id": schema.StringAttribute{
+				MarkdownDescription: "The unique identifier of the collection (scope) for the dashboard variable.",
+				Required:            true,
+			},
+			"default_object_selection": schema.StringAttribute{
+				MarkdownDescription: "The default object selection for the dashboard variable. Allowed values: `none`, `all`.",
+				Required:            true,
+				Validators: []validator.String{stringvalidator.OneOf(
+					"none",
+					"all",
+				)},
+			},
+			"allow_multiple_object_selection": schema.BoolAttribute{
+				MarkdownDescription: "Whether to allow multiple object selection for the dashboard variable.",
+				Optional:            true,
+				Computed:            true,
+				Default:             booldefault.StaticBool(false),
+			},
+			"last_updated": schema.StringAttribute{
+				MarkdownDescription: "Last updated timestamp of the dashboard variable",
+				Computed:            true,
+			},
+		},
+	}
+}
+
+func (r *DashboardVariableResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*SquaredUpClient)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Configure Type",
+			"Expected SquaredUpClient, got something else.",
+		)
+		return
+	}
+
+	r.client = client
+}
+
+func (r *DashboardVariableResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan squaredupDashboardVariable
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if diags.HasError() {
+		return
+	}
+
+	if plan.DefaultObjectSelection.ValueString() == "all" && plan.AllowMultipleObjectSelection.ValueBool() {
```

Comment:
Does our API not do this check/error? Shame to have to embed such UX/business logic here.

Reply:
Our API allows us to do that therefore I wrote this check to match the UX.